frontend/src/components/dashboard/tests/Dashboard.test.tsx

```diff
@@ -0,0 +1,893 @@
+import userEvent from '@testing-library/user-event'
+import React from 'react'
+import {
+  server,
+  render as _render,
+  screen,
+  fireEvent,
+  waitFor,
+  mockCommonApis,
+  TWILIO_CREDENTIAL,
+  TELEGRAM_CREDENTIAL,
+  DEFAULT_FROM,
+  CSV_FILENAME,
+} from 'test-utils'
+import Dashboard from '../Dashboard'
+
+const REPLY_TO = 'testEmail@open.gov.sg'
+const MESSAGE_TEXT = 'Test message'
+const UNPROTECTED_MESSAGE_TEXT = 'Test message {{protectedlink}}'
+const CAMPAIGN_NAME = 'Test campaign name'
+const SUBJECT_TEXT = 'Test subject'
+const RECIPIENT_EMAIL = 'testEmailRecipient@gmail.com'
+const RECIPIENT_NUMBER = '89898989'
+const PROTECTED_PASSWORD = 'test password'
+const EMAIL_CSV_FILE = new File(
+  [`recipient,password\n${RECIPIENT_EMAIL},${PROTECTED_PASSWORD}`],
+  CSV_FILENAME,
+  {
+    type: 'text/csv',
+  }
+)
+const MOBILE_CSV_FILE = new File(
+  [`recipient\n${RECIPIENT_NUMBER}`],
+  CSV_FILENAME,
+  {
+    type: 'text/csv',
+  }
+)
+
+function mockApis() {
+  const { handlers } = mockCommonApis({
+    curUserId: 1, // Start out authenticated; 1-indexed
+  })
+  return handlers
+}
+
+const renderDashboard = () =>
+  _render(<Dashboard />, {
+    router: { initialIndex: 0, initialEntries: ['/campaigns'] },
+  })
+
+test('successfully creates and sends a new email campaign', async () => {
+  // Setup
+  jest.useFakeTimers()
+  server.use(...mockApis())
+  renderDashboard()
```

Comment:
Could this be part of beforeEach() since they are the same for all the tests?

Reply:
Yup, but I'm a bit hesitant in doing so, because I'm not certain if all Dashboard tests in the future will definitely use these lines for setup.